### DISCUSSION TITLE
Invalid Payment Intent client secret

### DISCUSSION THREAD

**Discussion by dk-dsndr27** [flutter-stripe/flutter_stripe#1693] [Q&A]
Hi @remonh87,

can you help me, i got this error when i try for Stripe.instance.presentPaymentSheet().

Throw:
Exception has occurred.
_$StripeExceptionImpl (StripeException(error: LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: Invalid Payment Intent client secret: {{client_secret}}, message: Invalid Payment Intent client secret: {{client_secret}}, stripeErrorCode: null, declineCode: null, type: null)))

Debug Console:
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: StripeException(error: LocalizedErrorMessage(code: FailureCode.Failed, localizedMessage: Invalid Payment Intent client secret: {{client_secret}}, message: Invalid Payment Intent client secret:  {{client_secret}}, stripeErrorCode: null, declineCode: null, type: null))

And for the Publishable key and Secret key, i think same with my Stripe account

Thank's :)

**Comment by remonh87** [comment]
you need to make sure that you add the right paymentintent secret to the paymentsheet. See for example: https://docs.page/flutter-stripe/flutter_stripe/sheet

**Comment by iZenrix** [comment]
@dk-dsndr27  Have you solved the problem? I encountered the same error as you did.

**Reply by dk-dsndr27** [reply]
hi @iZenrix 

sure,
have you check your keys on your https://dashboard.stripe.com/test/apikeys ?
and the first think, you have to check the customer with user email

...
dynamic cust;
ProfileResponse profile = await ProfileService.getProfile();
dynamic checkCust = await SubscriptionService.checkCustomer(email: profile.email!);

log("${checkCust["data"]}");

if (checkCust["data"].isEmpty) {
  cust = await SubscriptionService.setCustomer(
    email: profile.email!,
    name: profile.email!,
  );
} else {
  cust = checkCust["data"][0];
}
...

**Reply by iZenrix** [reply]
@dk-dsndr27 Thanks for the suggestion, I will try to implement it.
